modules/perception/camera/lib/lane/detector/darkSCNN/darkSCNN_lane_detector.cc

```diff
@@ -0,0 +1,272 @@
+/******************************************************************************
+* Copyright 2018 The Apollo Authors. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*****************************************************************************/
+#include "modules/perception/camera/lib/lane/detector/darkSCNN/darkSCNN_lane_detector.h"
+
+#include <algorithm>
+#include <map>
+
+#include "modules/common/util/file.h"
+#include "modules/perception/camera/common/util.h"
+#include "modules/perception/inference/inference_factory.h"
+#include "modules/perception/inference/utils/resize.h"
+
+namespace apollo {
+namespace perception {
+namespace camera {
+
+using apollo::cyber::common::GetAbsolutePath;
+
+bool DarkSCNNLaneDetector::Init(const LaneDetectorInitOptions &options) {
+  std::string proto_path = GetAbsolutePath(options.root_dir, options.conf_file);
+  if (!apollo::common::util::GetProtoFromFile(proto_path, &darkscnn_param_)) {
+    AINFO << "load proto param failed, root dir: " << options.root_dir;
+    return false;
+  }
+  std::string param_str;
+  google::protobuf::TextFormat::PrintToString(darkscnn_param_, &param_str);
+  AINFO << "darkSCNN param: " << param_str;
+
+  const auto model_param = darkscnn_param_.model_param();
+  std::string model_root =
```

Comment:
Apply 'const' as often as possible.

Reply:
I am following the convention of other existing code to use const here. Prefer to keep this way. But I will keep it in mind.